design/httpproxy-status-meta.md

```diff
@@ -0,0 +1,52 @@
+# HTTPProxy Status Metadata
+
+Status: Draft
+
+## Abstract
+Since the addition of `Includes` for HTTPProxy, it can be difficult for owners of child HTTPProxy resources to understand what has been included to them without having access to a parent HTTPProxy.
+ 
+Users should have a way to understand this extra metadata about the ingress resources. 
+In addition, with this information, additional tooling can be developed against HTTPProxy resources to drive dashboards (i.e. Octant) and other `kubectl plugins`.
+
+## Goals
+- Expose extra metadata information to HTTPProxy resources
+
+## Detailed Design
+Extend the HTTPProxy.Status struct to include additional fields to support exposing metadata information:
+
+```
+type HTTPProxyStatus struct {
+  ...
+  <existing fields>
+  ...
+  
+  Metadata HTTPProxyMetadata `json:"metadata,omitempty"`
+}
+
+type HTTPProxyMetadata struct {
+  // ParentsRef shows one level parent of any HTTPProxy resources 
+  // which have included this resource.
+  Includes []HTTPProxyParent `json:"includes,omitempty"`
+}
+
+// HTTPProxyParent describes an HTTPProxy who included
+// any MatchConditions or Fqdn to a child HTTPProxy.
+type HTTPProxyParent struct {
```

Comment:
Perhaps in here instead of just `name` and `namespace` we should use one of the `ObjectReference` fields? Maybe that would make consuming this information with tooling easier.

Reply:
So just tuck this into another struct like `GatewayReference`? (https://kubernetes-sigs.github.io/service-apis/spec/)